manifest_staging/charts/gatekeeper/crds/assign-customresourcedefinition.yaml

```diff
@@ -4,7 +4,10 @@ metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.3.0
   labels:
-    gatekeeper.sh/system: "yes"
```

Comment:
is dropping these labels intentional? Are they being replaced somehow?

Reply:
+1, we should keep these labels

Reply:
This was a result of some gaps in the yaml generation pipeline.  More specifically, two mutation related CRDs were not included in `config/crd/kustomization.yaml`.  As mutation wasn't ready for primetime, this made sense.  This kept those CRDs out of the final `gatekeeper.yaml`.

The `make manifests` target calls `kustomize build` on the `cmd/build/helmify` directory, which relies on a long chain of dependent `kustomization.yaml` files that include the mutation CRDs.  As the mutation CRDs were left out of the `kustomization.yaml` in their original folder (`config/crd/bases/`), these CRDs were manually copied into `config/overlays/mutation_webhook/`.  This solved the problem temporarily, but left a footgun for me as I upgraded `controller-gen`.

The solution was to automate this copy step, which I have now done.  That removes the "log jam" in the generation pipeline, renewing the generation of CRDs that was already broken.  We simply did not notice because there was no situation that surfaced this problem.